packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js

```diff
@@ -0,0 +1,440 @@
+                dataSourceEntityFilter: {
+                  attributes_type: `${attributeFilter}`,
+                },
+              },
+            },
+          ],
+        },
+      },
+    ],
+    defaultValues: {
+      sum: 0,
+      count: 0,
+    },
+  },
+});
+
+const generateTargetIndicator = (code, dataElement, target) => ({
+  id: generateId(),
+  code,
+  builder: 'analyticArithmetic',
+  config: {
+    formula: `${target}`,
+    aggregation: { [dataElement]: 'FINAL_EACH_YEAR' },
+  },
+});
+
+const generateFiveYearIndicator = (code, dataElement, target) => ({
+  id: generateId(),
+  code,
+  builder: 'analyticArithmetic',
+  config: {
+    formula: `${dataElement} >= ${target} ? 1 : ${dataElement} >= firstExistingValue(value1YearAgo, value2YearsAgo, value3YearsAgo, value4YearsAgo, value5YearsAgo, Infinity) ? 0 : -1`,
+    parameters: [
+      {
+        builder: 'analyticArithmetic',
+        // 1 year is defined separately because I don't want to handle singular/plural year(s)
+        code: 'value1YearAgo',
+        config: {
+          formula: `${dataElement}`,
+          aggregation: [
+            'FINAL_EACH_YEAR',
+            {
+              type: 'OFFSET_PERIOD',
+              config: {
+                offset: 1,
+                periodType: 'year',
+              },
+            },
+          ],
+        },
+      },
+      ...[2, 3, 4, 5].map(year => ({
+        builder: 'analyticArithmetic',
+        code: `value${year}YearsAgo`,
+        config: {
+          formula: `${dataElement}`,
+          aggregation: [
+            'FINAL_EACH_YEAR',
+            {
+              type: 'OFFSET_PERIOD',
+              config: {
+                offset: year,
+                periodType: 'year',
+              },
+            },
+          ],
+        },
+      })),
+    ],
+    aggregation: 'FINAL_EACH_YEAR',
+    defaultValues: {
+      value1YearAgo: 'undefined',
+      value2YearsAgo: 'undefined',
+      value3YearsAgo: 'undefined',
+      value4YearsAgo: 'undefined',
+      value5YearsAgo: 'undefined',
+    },
+  },
+});
+
+const DRILL_DOWN_REPORT_CONFIG = {
+  fetch: {
+    dataElements: [
+      'csr_priority_district_average',
+      'csr_non_priority_district_average',
+      'csr_priority_district_target',
+      'csr_non_priority_district_target',
+    ],
+    aggregations: [
+      {
+        type: 'FINAL_EACH_YEAR',
+      },
+    ],
+  },
+  transform: [
+    {
+      transform: 'insertColumns',
+      columns: {
+        "=translate($dataElement, { csr_priority_district_average: 'Priority Districts Average', csr_non_priority_district_average: 'Other Districts Average',  csr_priority_district_target: 'Priority Districts Target', csr_non_priority_district_target: 'Other Districts Target' })":
+          '=$value/100',
+        timestamp: '=periodToTimestamp($period)',
+      },
+    },
+    {
+      transform: 'mergeRows',
+      groupBy: 'timestamp',
+      using: 'last',
```

Comment:
```suggestion
```
Default is now `single`, which adds the extra assertion that we are only expecting one to be defined